packages/console/src/pages/SignInExperience/components/LanguagesForm.tsx

```diff
@@ -23,9 +24,17 @@ const LanguagesForm = ({ isManageLanguageVisible = false }: Props) => {
   const { watch, control, register } = useFormContext<SignInExperienceForm>();
   const isAutoDetect = watch('languageInfo.autoDetect');
   const [isManageLanguageFormOpen, setIsManageLanguageFormOpen] = useState(false);
+  const { languages } = useUiLanguages();
 
-  const { context: customPhrasesContext, Provider: CustomPhrasesContextProvider } =
-    useCustomPhrasesContext();
+  const languageOptions = useMemo(() => {
+    return languages.map((languageTag) => ({
+      value: languageTag,
+      title: uiLanguageNameMapping[languageTag],
+    }));
+  }, [languages]);
+
+  const { context: languageEditorContext, Provider: LanguageEditorContextProvider } =
+    useLanguageEditorContext(languages);
```

Comment:
nit: This doesn't necessarily have to be encapsulated into a hook.

Reply:
I think it's ok here.
It keeps all related things together, just refer to the [`usePageContext`](https://github.com/logto-io/logto/blob/master/packages/ui/src/hooks/use-page-context.ts).